walk/walk.go

```diff
@@ -182,7 +191,7 @@ func visit(c *config.Config, cexts []config.Configurer, knownDirectives map[stri
 			subdirs = append(subdirs, base)
 
 			if updateRels.shouldVisit(entRel, shouldUpdate) {
-				visit(c, cexts, knownDirectives, updateRels, t, wf, entRel, shouldUpdate)
+				visit(c.Clone(), cexts, knownDirectives, updateRels, t, wf, entRel, shouldUpdate)
```

Comment:
We would now clone once per child rather than once, right? That sounds worse in the abstract, have you measured the performance impact?

Reply:
It's the exact same, just less confusing imo. It creates a root `Config` and then clones once per visited child directory.

Previously:
1. construct the root `Config` and pass it to the root `visit` invocation
2. pass that config into `configure` which *might* clone it and return a new one
3. pass the `configure`-returned config into the recursive `visit`

In step 1 the config might be for the parent directory or the current directory which is super confusing.
From step 2 onward it is the config of the current directory due to the conditional cloning within `configure`.

IMO neither of those are clear and it confuses me every time (including right now trying to remember+describe what's going on).

Now:
1. construct the root `Config` and pass it to the root `visit` invocation
2. `configure` is passed the config of the directory being configured
3. clone the config for each child when recursing

The config passed to `visit` is for the directory being visited.
The config passed to `configure` is the config for the directory being configured.
No more confusion about the config being for the parent vs current directory, no more conditional cloning, no more switching from parent to current dir config in the middle of the method etc.

Reply:
Thanks for the explanation! I now understand why this is simpler.

My only concern is that "possibly cloned" can mean "may optimize out a clone", so I would like to see @tyler-french or @linzhp run this in a benchmark.

Reply:
When I say "possibly cloned" I actually mean "cloned if not the root". So we construct the root config, then clone per child. Same thing after my change except without all the confusing (imo) conditional logic.

Reply:
Ah, right, that makes this an easy choice.